certificateshop-logic/src/main/java/com/epam/esm/dao/impl/jpa/JpaOrderDaoImpl.java

```diff
@@ -0,0 +1,220 @@
+            " (userOrderId, certificateId, certificateInJSON, certificatePrice) values (?, ?, ?, ?)";
+    private static final List<String> ORDER_HEADERS = Arrays.asList(ColumnNames.TABLE_USER_COLUMN_ID,
+            ColumnNames.TABLE_USER_COLUMN_NICKNAME, ColumnNames.TABLE_USERORDER_COLUMN_ID,
+            ColumnNames.TABLE_USERORDER_COLUMN_CREATE_DATE, ColumnNames.TABLE_USERORDER_COLUMN_NAME,
+            ColumnNames.TABLE_USERORDER_CERTIFICATE_COLUMN_CERTIFICATEINJSON);
+    private OrderRepository orderRepository;
+    private EntityManager entityManager;
+    private ListToSetConverter listToSetConverter;
+    private OrderExtractor orderExtractor;
+
+    @Autowired
+    public JpaOrderDaoImpl(OrderRepository orderRepository, EntityManager entityManager,
+                           ListToSetConverter listToSetConverter, OrderExtractor orderExtractor) {
+        this.orderRepository = orderRepository;
+        this.entityManager = entityManager;
+        this.listToSetConverter = listToSetConverter;
+        this.orderExtractor = orderExtractor;
+    }
+
+    /**
+     * Saves {@link Order} in the database.
+     *
+     * @param order is the {@link Order} to save.
+     */
+    @Override
+    public void save(Order order) {
+        entityManager
+                .createNativeQuery(INSERT_ENTITY_SQL)
+                .setParameter(1, order.getUser().getId())
+                .setParameter(2, order.getCreateDate())
+                .setParameter(3, order.getName())
+                .executeUpdate();
+    }
+
+    /**
+     * Finds all {@link Order} entity in the database.
+     *
+     * @return List of the {@link Order} objects.
+     */
+    @Override
+    public List<Order> findAll() {
+        Query query = entityManager.createNativeQuery(FIND_ALL_ENTITIES_SQL);
+        List<Object[]> resultList = query.getResultList();
+        List<List<Object>> result = convertListOfArrayToListOfLists(resultList);
+        return getEntities(result);
+    }
+
+    /**
+     * Finds {@link Optional <Order>} in the database by the id of the {@link Order}.
+     *
+     * @param id is the {@link long} to find.
+     * @return {@link Optional<Order>}.
+     */
+    @Override
+    public Optional<Order> findById(long id) {
+        Query query = entityManager.createNativeQuery(FIND_ENTITY_BY_ID_SQL).setParameter(1, id);
+        List<Object[]> resultList = query.getResultList();
+        List<List<Object>> result = convertListOfArrayToListOfLists(resultList);
+        List<Order> orders = getEntities(result);
+        return orders.stream().findFirst();
+    }
+
+    /**
+     * Updates the {@link Order}.
+     *
+     * @param order is the value of the {@link Order} to update.
+     */
+    @Override
+    public void update(Order order) {
+
+    }
+
+    /**
+     * Removes the {@link Order} object from the database.
+     *
+     * @param id is the value of the {@link long} to find.
+     */
+    @Override
+    public void delete(long id) {
+
+    }
+
```

Comment:
Why do you keep such empty methods? What do you know about SOLID (exactly I letter)?

Reply:
Because i use this methods in the jdbcDao. And they are the part of the interface and should be overriden in the new realisation.
Yes, interface segregation should be done. Just now i have no idea how to implement it.

Reply:
done